service/matching/taskReader.go

```diff
@@ -297,6 +296,35 @@ func (tr *taskReader) persistAckLevel() error {
 	return nil
 }
 
-func (tr *taskReader) isTaskAddedRecently(lastAddTime time.Time) bool {
-	return time.Since(lastAddTime) <= tr.config.MaxTasklistIdleTime()
+// completeTask marks a task as processed. Only tasks created by taskReader (i.e. backlog from db) reach
+// here. As part of completion:
+//   - task is deleted from the database when err is nil
+//   - new task is created and current task is deleted when err is not nil
+func (tr *taskReader) completeTask(task *persistence.TaskInfo, err error) {
```

Comment:
Chatted offline about moving this from `taskListManager` to `taskReader`. The plan is to refactor this further so that this is `handleError` that remains in `taskReader` and move the ack level update logic back to the `taskListManager`. It'll be done in a separate PR.

Reply:
Chatted offline, the ack level update logic will still stay in `taskReader`, but we will split the completeTask function to `AckTask` and `NackTask`